### ISSUE TITLE
Calculix 3D beam B31 and B32 possible for comparison of beam elements

### ISSUE THREAD

**Issue by stefano2734** [Krande/adapy#102]
**Is your feature request related to a problem? Please describe.**

see http://www.dhondt.de/ccx_2.21.pdf
page 118 and 119

B31 and B32 are the elements for this calculation with beams in Calculix ccx.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by Krande** [comment]
Hey @stefano2734! 

Would you like me to add these beam elements for the Calculix ada-py FEM tests?
If I recall correcly, the reason for not including these elements were because Calculix did not support eigenvalue analysis using these element types with the specific cross section. I can't recall exactly the error message I got when attempting to solve this, but I vaguely recall something about it being possible using either a different custom element type and/or using general sections.

If you have any advice on this, I would appreciate any help you can provide:)

Let me know if I've misunderstood your comment?

Best Regards
Kristoffer

**Comment by stefano2734** [comment]
I made a test with userbeam.inp: no modal analyze with u1 elements.
So a change to other beam element like b31 or b32 or others in calculix is necessary.

**Comment by Krande** [comment]
Right, and if I remember correctly there are a limited number of cross section types supported for calculix B31/B32 elements? (just pipe and box if I am not mistaken?)

So if we should switch to B31/B32 we would have to modify the cross section properties to match that of any other cross section? 

I believe that is the core reason I ended up using the general section element type. It just seemed easier to directly input the cross sectional properties such as A, I11, I12, I22 etc.. 

Maybe it's possible to expand the number of supported cross section types beyond just pipe and box by writing custom types? (I don't have any experience with that though :( )

**Comment by stefano2734** [comment]
beam section of b31 and b32 is limited in documentation to rectangular, elliptical, pipe and box without general. 
https://www.feacluster.com/CalculiX/ccx_2.18/doc/ccx/node60.html
with internal expansion to  C3D8I or C3D20  elements it should be solvable in future by the developers of calculix.
but I will check this. 
sometimes the documentation is not actual.

another alternative is the Uxxxx Element with own values to beam element with section general for super advanced users.
https://www.feacluster.com/CalculiX/ccx_2.18/doc/ccx/node72.html

**Comment by stefano2734** [comment]
A fast workaround for beam elements b31 or b32 here is three rectangular sections for Doppel T.

here the example of a composite cantilever beam with two rectangular sections.
https://feacluster.com/CalculiX/ccx_2.18/doc/ccx/input_deck_viewer.php?input_deck=beamcom.inp

**Comment by stefano2734** [comment]
A modified calculix 2.23 with extended beam can  perhaps the double T  beam problem.
A user beam element is here implemented.
Perhaps this beam will implement in next official calculix version.

https://github.com/Raztua/CCX-UB21
